google-cloud-datastore/src/main/java/com/google/cloud/datastore/ReadOptionProtoPreparer.java

```diff
@@ -0,0 +1,77 @@
+/*
```

Comment:
This is the only preparer that is a bit confusing to me - where is this logic extracted from?

If this is not a refactor, we should do this as a feature separately. This PR should focus only on the refactoring of code.

Reply:
This is extracted from DatastoreImpl.java [this location to be precise](https://github.com/googleapis/java-datastore/pull/876/files#diff-f6b1d8b631b8df3355d5820957585afd4ae2fb5df43357e10dc29828746c2fdfL340-L367).

It's an old code which is extracted to this ProtoPreparer, so that it can be reused in AggregationQuery in count feature.

Reply:
Got it. I still see some inconsistencies - can you confirm these are intended?

1. https://github.com/googleapis/java-datastore/pull/876/files#diff-f6b1d8b631b8df3355d5820957585afd4ae2fb5df43357e10dc29828746c2fdfL346-L350. This check appears to be missing
2. https://github.com/googleapis/java-datastore/pull/876/files#diff-1a25d0e25821474babf2fe4712c0895b18dbac81f3b0883581cb58bf601693d2R59-R65 This seems new

Reply:
Also https://github.com/googleapis/java-datastore/pull/876/files#diff-1a25d0e25821474babf2fe4712c0895b18dbac81f3b0883581cb58bf601693d2R42-R46 seems new

Reply:
Taking a closer look, I see what the intention is by first checking for more than one key. What happens if you set both eventual consistency and transaction id?

Reply:
Discussed offline, readoptions can only have one consistency type; see https://github.com/googleapis/java-datastore/blob/main/proto-google-cloud-datastore-v1/src/main/proto/google/datastore/v1/datastore.proto#L462-L499